File: 821. Shortest Distance to a Character.py
#PYTHON
#* 821. Shortest Distance to a Character
s = "loveleetcode"; c = "e"
from typing import List
#*
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        res = []
        char_indices = [i for i, char in enumerate(s) if char == c]
        
        for i in range(len(s)):
            closest = min(char_indices, key=lambda x: abs(x - i))
            res.append(abs(i - closest))
        
        return res
    
print(Solution().shortestToChar(s,c))

#*
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
      res = []
      char_indices = [i for i, char in enumerate(s) if char == c]
      
      for i in range(len(s)):
          res.append(min(abs(i - index) for index in char_indices))
      
      return res
    
print(Solution().shortestToChar(s,c))

        

File: 821. Shortest Distance to a Character.rs
// 821. Shortest Distance to a Character
//s = "loveleetcode"; c = "e"
impl Solution {
  pub fn shortest_to_char(s: String, c: char) -> Vec<i32> {
      let mut res = Vec::new();
      let chars: Vec<char> = s.chars().collect();
      let nums: Vec<usize> = chars.iter().enumerate().filter_map(|(i, &ch)| if ch == c { Some(i) } else { None }).collect();

      for (i, _) in chars.iter().enumerate() {
          let closest = nums.iter().map(|&x| (x as i32 - i as i32).abs()).min().unwrap();
          res.push(closest);
      }
      res
  }
}
